code/modules/heavy_vehicle/mech_interaction.dm

```diff
@@ -565,3 +541,163 @@
 		src.visible_message("<span class='warning'>\The [src] beeps loudly as its servos sieze up, and it enters lockdown mode!</span>")
 	else
 		src.visible_message("<span class='warning'>\The [src] hums with life as it is released from its lockdown mode!</span>")
+
+/mob/living/heavy_vehicle/get_floating_chat_x_offset()
+	return 8
+
+/mob/living/heavy_vehicle/hear_say(var/message, var/verb = "says", var/datum/language/language = null, var/alt_name = "", var/italics = 0, var/mob/speaker = null, var/sound/speech_sound, var/sound_vol)
+	if(can_listen())
+		handle_hear_say(speaker, message)
+	return ..()
+
+// heavily commented so it doesn't look like one fat chunk of code, which it still does - Geeves
+/mob/living/heavy_vehicle/proc/handle_hear_say(var/mob/speaker, var/text)
+	var/found_text = findtext(text, name)
+	if(!found_text)
+		found_text = findtext(text, nickname)
+	if(found_text)
+		text = copytext(text, found_text) // I'm trimming the text each time so only information stated after eachother is valid
+		sleep(5)
+
+		// a quick way to figure out the remote control status of the mech
+		if(findtext(text, "report diagnostics"))
+			var/has_leader = FALSE
+			if(leader)
+				var/mob/resolved_leader = leader.resolve()
+				if(!resolved_leader)
+					say("Error, leader not found. Unassigning...")
+					unassign_leader()
+					return
+				has_leader = TRUE
+			say("Currently [has_leader ? "paired with [leader_name]" : "unpaired"].")
+			if(following)
+				var/mob/resolved_following = following.resolve()
+				if(!resolved_following)
+					say("Error, follow target not found. Unassigning...")
+					unassign_following()
+				else
+					say("Currently following [resolved_following.name].")
+			if(nickname)
+				say("Nickname set to [nickname].")
+			say("Maintenance protocols, [maintenance_protocols ? "active" : "disabled"].")
+			return
+
+		// Checking whether we have a leader or not
+		if(!leader)
+			if(!maintenance_protocols) // don't select a leader unless we have maintenance protocols set
+				return
+			// If we have no leader, we listen to the keywords 'listen to'
+			if(findtext(text, "listen to"))
+				text = copytext(text, found_text)
+				found_text = findtext(text, "me") // if they say listen to me, we listen to them
+				if(found_text)
+					assign_leader(speaker)
+					say("New paired leader, [leader_name], confirmed and added to temporary biometric database.")
+					return
+				// check for humans and their IDs
+				var/list/humans_in_view = list()
+				for(var/mob/living/carbon/human/H in view(world.view, src))
+					var/obj/item/card/id/ID = H.GetIdCard(TRUE)
+					if(ID?.registered_name) // we ID people based on their... ID
+						humans_in_view[ID.registered_name] = H
+				for(var/h in humans_in_view)
+					if(findtext(text, h))
+						var/mob/living/carbon/human/H = humans_in_view[h]
+						assign_leader(H)
+						say("New paired leader, [h], confirmed and added to temporary biometric database.")
+						break
```

Comment:
I don't see a reason why these loops are not combined.

Reply:
combined 'em